src/main/java/ch/njol/skript/expressions/ExprAge.java

```diff
@@ -0,0 +1,172 @@
+public class ExprAge extends SimplePropertyExpression<Object, Integer> {
+
+	static {
+		String fromType = "";
+		
+		if (Skript.classExists("org.bukkit.entity.Ageable"))
+			fromType += "entities";
+		if (Skript.classExists("org.bukkit.block.data.Ageable"))
+			fromType += "/blocks"; // org.bukkit.entity.Ageable exists before org.bukkit.block.data.Ageable
+
+		if (!fromType.isEmpty())
+			register(ExprAge.class, Integer.class, "[:max[imum]] age", fromType);
+	}
+
+	private boolean isMax = false;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		isMax = parseResult.hasTag("max");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Object obj) {
+		if (obj instanceof Block) {
+			BlockData bd = ((Block) obj).getBlockData();
+			if (!(bd instanceof Ageable))
+				return null;
+			Ageable ageable = (Ageable) bd;
+			return isMax ? ageable.getMaximumAge() : ageable.getAge();
+		} else if (obj instanceof org.bukkit.entity.Ageable) {
+			return  ((org.bukkit.entity.Ageable) obj).getAge();
+		}
+		return null;
+	}
+	
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (isMax || mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.DELETE)
+			return null;
+		return CollectionUtils.array(Number.class);
+
+	}
+	
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (mode != ChangeMode.RESET && mode != ChangeMode.DELETE && (delta == null || delta[0] == null))
+			return;
+
+		int newValue;
+		switch (mode) {
+			case ADD:
+				newValue = ((Number) delta[0]).intValue();
+				for (Object obj : getExpr().getArray(event)) {
+					Number oldValue = convert(obj);
+					if (oldValue == null)
+						continue;
+					setAge(obj, oldValue.intValue() + newValue);
+				}
+				break;
+			case REMOVE:
+				newValue = ((Number) delta[0]).intValue();
+				for (Object obj : getExpr().getArray(event)) {
+					Number oldValue = convert(obj);
+					if (oldValue == null)
+						continue;
+					setAge(obj, oldValue.intValue() - newValue);
+				}
+				break;
+			case SET:
+				newValue = ((Number) delta[0]).intValue();
+				for (Object obj : getExpr().getArray(event)) {
+					setAge(obj, newValue);
+				}
+				break;
+			case RESET:
+				for (Object obj : getExpr().getArray(event)) {
+					int value = 0;
+					// baby animals takes 20 minutes to grow up - ref: https://minecraft.fandom.com/wiki/Breeding
+					if (obj instanceof org.bukkit.entity.Ageable)
+						// it might change later on so removing entity age reset would be better unless
+						// bukkit adds a method returning the default age
+						value = -24000;
+					setAge(obj, value);
+				}
+				break;
+		}
+	}
+	
+	@Override
+	public Class<? extends Integer> getReturnType() {
+		return Integer.class;
+	}
+	
+	@Override
+	protected String getPropertyName() {
+		return (isMax ? "max " : "") + "age";
```

Comment:
```suggestion
		return (isMax ? "maximum " : "") + "age";
```
same for other stuff, such as docs. We should use proper English here